ui/packages/consul-ui/app/services/repository/metrics.js

```diff
@@ -17,16 +17,17 @@ export default class MetricsService extends RepositoryService {
 
   init() {
     super.init(...arguments);
-    const config = this.config.get();
+    // TODO: this flow should be be async, then can just use either get or a DataSource
+    const config = this.config.getSync();
     // Inject whether or not the proxy is enabled as an option into the opaque
     // JSON options the user provided.
     const opts = config.metrics_provider_options || {};
     opts.metrics_proxy_enabled = config.metrics_proxy_enabled;
+    // Inject the base app URL
+    const provider = config.metrics_provider || 'prometheus';
```

Comment:
This needs to all be changed to use an asynchronous flow like all our other repositories. I figured I could do that later so explicitly called what it does now to `getSync` so its clear that it synchronous even though it should all be async. We can tackle this later on.